### PR TITLE
Inteon/moadeli/tip bump

### PR DESCRIPTION
**Pull request by mmoadeli**
The PR has all changes required to make Polygeist process input SYCL kernel. There are a number of commits which are here as work around. They are going to be addressed.

### CODE REVIEW

File: tools/mlir-clang/Lib/clang-mlir.cc [hunk 1]
```diff
@@ -4836,25 +4837,38 @@ static bool parseMLIR(const char *Argv0, std::vector<std::string> filenames,
 
   Argv.push_back("-emit-ast");
 
-  const unique_ptr<Compilation> compilation(
-      driver->BuildCompilation(llvm::ArrayRef<const char *>(Argv)));
-  JobList &Jobs = compilation->getJobs();
-  if (Jobs.size() < 1)
-    return false;
+llvm::SmallVector<const ArgStringList *, 4> CommandList;
```
Comment (joeatodd) [inline tools/mlir-clang/Lib/clang-mlir.cc:4860]: Indentation ->

File: tools/mlir-clang/Lib/clang-mlir.cc [hunk 2]
```diff
@@ -4836,25 +4837,38 @@ static bool parseMLIR(const char *Argv0, std::vector<std::string> filenames,
...
 
-    const ArgStringList *args = &cmd->getArguments();
+  MLIRAction Act(fn, module);
+
+  for (const ArgStringList *args : CommandList) {
+    std::unique_ptr<CompilerInstance> Clang(new CompilerInstance());
 
     Success = CompilerInvocation::CreateFromArgs(Clang->getInvocation(), *args,
-                                                 Diags);
+                                                  Diags);
```
Comment (joeatodd) [inline tools/mlir-clang/Lib/clang-mlir.cc:4891]: Indentation

File: tools/mlir-clang/CMakeLists.txt [hunk 3]
```diff
@@ -73,3 +73,8 @@ target_link_libraries(mlir-clang PRIVATE
 )
 add_dependencies(mlir-clang MLIRPolygeistOpsIncGen MLIRPolygeistPassIncGen)
 add_subdirectory(Test)
+Message(STATUS "Mahmoud clang dir: ${CLANG_DIR}")
```
Comment (joeatodd) [inline tools/mlir-clang/CMakeLists.txt:76]: Should this debug statement be tidied away?

File: tools/mlir-clang/mlir-clang.cc [hunk 4]
```diff
@@ -136,6 +132,12 @@ static cl::list<std::string> defines("D", cl::desc("defines"),
 static cl::list<std::string> Includes("include", cl::desc("includes"),
                                       cl::cat(toolOptions));
 
+static cl::list<std::string> inputFileName(cl::Positional, cl::OneOrMore,
+                                           cl::desc("<Specify input file>"),
+                                           cl::cat(toolOptions));
+
```
Comment (joeatodd) [inline tools/mlir-clang/mlir-clang.cc:138]: I don't know much about the LLVM CommandLine library - as I was reading the docs for `cl::ConsumeAfter` below, I [found](https://llvm.org/docs/CommandLine.html#the-cl-consumeafter-modifier) the following:

> There are several limitations to when cl::ConsumeAfter options can be specified. For example, only one cl::ConsumeAfter can be specified per program, there must be at least one [positional argument](https://llvm.org/docs/CommandLine.html#positional-argument) specified, **there must not be any [cl::list](https://llvm.org/docs/CommandLine.html#cl-list) positional arguments**, and the cl::ConsumeAfter option should be a [cl::list](https://llvm.org/docs/CommandLine.html#cl-list) option.

I think having `inputFileName` defined as a `cl::list` violates this?

**Review by joeatodd — APPROVED** [review]
LGTM aside from possible violation of requirements of `cl::ConsumeAfter` and a few minor formatting issues.

File: tools/mlir-clang/mlir-clang.cc (hunk 4 above)
Comment (mmoadeli) [reply tools/mlir-clang/mlir-clang.cc:138]: Thanks. That makes sense. I create  a task to have this investigated.

Comment (mmoadeli) [reply tools/mlir-clang/mlir-clang.cc:138]: https://jira.office.codeplay.com/browse/II-187

File: tools/mlir-clang/Lib/clang-mlir.cc [hunk 5]
```diff
@@ -4254,6 +4267,12 @@ bool MLIRASTConsumer::HandleTopLevelDecl(DeclGroupRef dg) {
       continue;
     }
 
+    //  if (fd->getIdentifier())
+    //    llvm::errs() << "Func name: " << fd->getName() << "\n";
+    //  llvm::errs() << "Func Body && Loc " << "\n";
+    //  fd->getBody()->dump();
+    //  fd->getLocation().dump(SM);
+
```
Comment (frengels) [inline tools/mlir-clang/Lib/clang-mlir.cc:4275]: should commented out code get removed?

File: tools/mlir-clang/mlir-clang.cc [hunk 6]
```diff
@@ -456,8 +466,8 @@ int main(int argc, char **argv) {
       return 4;
     }
     if (mlir::failed(mlir::verify(module.get()))) {
-      module->dump();
-      return 5;
+//      module->dump();
```
Comment (frengels) [inline tools/mlir-clang/mlir-clang.cc:469]: could a comment be added explaining why these are commented out and linking to the JIRA ticket

File: tools/mlir-clang/mlir-clang.cc [hunk 7]
```diff
@@ -470,11 +480,11 @@ int main(int argc, char **argv) {
...
-      if (!Opt0) {
-        optPM.addPass(mlir::createCanonicalizerPass());
-        optPM.addPass(mlir::createCSEPass());
-        pm.addPass(mlir::createInlinerPass());
-      }
+      // if (!Opt0) {
+      //   optPM.addPass(mlir::createCanonicalizerPass());
+      //   optPM.addPass(mlir::createCSEPass());
+      //   pm.addPass(mlir::createInlinerPass());
+      // }
```
Comment (frengels) [inline tools/mlir-clang/mlir-clang.cc:487]: same comment and JIRA ticket here

**Review by frengels — CHANGES_REQUESTED** [review]
Just minor style nitpicks, and for chasing things down in the future

File: tools/mlir-clang/mlir-clang.cc (hunk 6 above)
Comment (mmoadeli) [reply tools/mlir-clang/mlir-clang.cc:469]: There is a ticket created for this and have the link to the associated commit.

File: tools/mlir-clang/Lib/clang-mlir.cc (hunk 5 above)
Comment (mmoadeli) [reply tools/mlir-clang/Lib/clang-mlir.cc:4275]: I found this widely used when debugging. I'd prefer keeping it.

File: tools/mlir-clang/mlir-clang.cc (hunk 6 above)
Comment (breanna-devore-mcdonald) [reply tools/mlir-clang/mlir-clang.cc:469]: @mmoadeli can that ticket link be added as a comment in the code?

**Review by breanna-devore-mcdonald — CHANGES_REQUESTED** [review]
The CI tests are failing- should be fixed before merging?

File: tools/mlir-clang/mlir-clang.cc (hunk 6 above)
Comment (breanna-devore-mcdonald) [reply tools/mlir-clang/mlir-clang.cc]: Saw the changes- looks good!

**Review by breanna-devore-mcdonald — APPROVED** [review]
The CI checks can be ignored- the issues are coming from the original repo, not any Codeplay commits

File: tools/mlir-clang/Lib/clang-mlir.cc [hunk 8]
```diff
@@ -4479,6 +4639,11 @@ mlir::Type MLIRASTConsumer::getMLIRType(clang::QualType qt, bool *implicitRef,
       }
     }
 
+    if (ST->getName().contains("class.cl::sycl") ||
```
Comment (keryell) [inline tools/mlir-clang/Lib/clang-mlir.cc:4642]: In SYCL 2020 the namespace is now `sycl` instead of `cl::sycl`.

**Pull request merged**